### ISSUE TITLE
Support explicit deriving strategies

### ISSUE THREAD

**Issue by kozross** [Benjmhart/vim-instantinstance#1]
It would be nice if this plugin could also support explicit deriving strategies (for example, it would spit out `deriving stock (Show)`.

**Comment by Benjmhart** [comment]
i thought about this and fully support it,   but i wanted to make sure the input methods and general API were ironed out a bit first - and that it was compatible for other users since i slapped this together and tested in neovim for 10 minutes before posting.

what would you suggest? something like `<leader>di` to do the same thing but with a deriving clause and a prompt for the strategy ([N]ewtype, [S]tock,  [A]nyclass, [V]ia) - in the event of via i suppose an additional prompt is needed for the via type.

does that sum it up?

**Comment by kozross** [comment]
Yeah, more or less. You can probably skip `via` - it's complex enough to deserve a full type-out. The others, however, are purely mechanical and should be automated away.

**Comment by Benjmhart** [comment]
something i spotted when looking at this is - i probably need to do some
correction around the round braces - this is a particularly bad behaviour
for purescript


Benjamin James Mikel Hart
